src/runtime/type-from-literal.ts

```diff
@@ -17,7 +17,7 @@ import {SlotInfo} from './slot-info.js';
 function fromLiteral(literal: TypeLiteral) : Type {
   switch (literal.tag) {
     case 'Entity':
-      return new EntityType(Schema.fromLiteral(literal.data));
+      return new EntityType(Schema.fromLiteral(literal.data), literal.refinement);
```

Comment:
`literal.data` is where the data is stored (this should include the refinement).

Lets add a round trip test for to/fromLiteral?